x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Chunk.java

```diff
@@ -130,11 +159,17 @@ protected TypeResolution resolveType() {
         if (childrenResolved() == false) {
             return new TypeResolution("Unresolved children");
         }
-        return isString(field(), sourceText(), FIRST).and(Options.resolve(options, source(), SECOND, ALLOWED_OPTIONS, this::verifyOptions));
+
+        TypeResolution resolution = isString(field(), sourceText(), FIRST);
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+
+        return Options.resolve(options(), source(), SECOND, ALLOWED_OPTIONS, this::verifyOptions);
     }
 
     private void verifyOptions(Map<String, Object> optionsMap) {
-        if (options == null) {
+        if (options() == null) {
```

Comment:
we have this in a bunch of places where we change to `options()` and `field()` - it's not a pattern you see in other implementations of ES|QL functions.
these are all instance variables referred in instance methods, we don't need to use the getter here.